### PR TITLE
Regress performance: Use `Arc<usize>` for more sudoku realism

### PR DESCRIPTION
**Pull request by konstin**
and have codspeed catch the regression.

### CODE REVIEW

**Comment by konstin** [comment]
This is a good example to why the sudoku benchmark is very unrealistic (we can silence it from reporting) and why we need to port realistic examples:

image

**Comment by Eh2406** [comment]
This is really cool! Thanks for the demonstration.

File: .github/workflows/benchmarks.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: Benchmarks (CodSpeed)
 on:
   push:
     branches:
-      - main
+      - dev
```
Comment (konstin) [inline .github/workflows/benchmarks.yml:6]: @Eh2406 i used the wrong branch name, that's why we currently don't get data from ~~main~~ dev.

**Comment by Eh2406** [comment]
I didn't realize this was something you actually wanted merged, I thought it was just a proof of what would happen if we regress performance. Merging now.

**Review by Eh2406 — APPROVED** [review]

**Comment by konstin** [comment]
I had changed my mind :)

The `Arc`ing made the sudoku more realistic, so I discarded the smallvec change.

**Pull request merged**